src/cswinrt/code_writers.h

```diff
@@ -1035,12 +1052,285 @@ remove => %.% -= value;
+        write_event(w, event.Name(), event, interface_member, visibility, ""sv, platform_attribute);
 
         if (is_overridable || !is_exclusive_to(event.Parent()))
         {
-            write_event(w, w.write_temp("%.%", bind<write_type_name>(event.Parent(), typedef_name_type::CCW, false), event.Name()), event, "this");
+            write_event(w, w.write_temp("%.%", bind<write_type_name>(event.Parent(), typedef_name_type::CCW, false), event.Name()), event, "this", ""sv, ""sv, platform_attribute);
+        }
+    }
+
+    auto write_custom_attribute_args(writer& w, CustomAttribute const& attribute, CustomAttributeSig const& signature)
+    {
+        auto write_fixed_arg = [&](writer & w, FixedArgSig arg)
+        {
+            if (std::holds_alternative<std::vector<ElemSig>>(arg.value))
+            {
+                throw_invalid("ElemSig list unexpected");
+            }
+            auto&& arg_value = std::get<ElemSig>(arg.value);
+
+            call(arg_value.value,
+                [&](ElemSig::SystemType system_type)
+                {
+                    auto arg_type = attribute.get_cache().find_required(system_type.name);
+                    if (is_static(arg_type))
+                    {
+                        w.write("typeof(%)", bind<write_projection_type>(arg_type));
+                    }
+                    else
+                    {
+                        w.write("typeof(%)", bind<write_projection_ccw_type>(arg_type));
+                    }
+                },
+                [&](ElemSig::EnumValue enum_value)
+                {
+                    if (enum_value.type.m_typedef.TypeName() == "AttributeTargets")
+                    {
+                        std::vector<std::string> values;
+                        auto value = std::get<uint32_t>(enum_value.value);
+                        if (value == 4294967295)
+                        {
+                            values.emplace_back("All");
+                        }
+                        else
+                        {
+                            static struct
+                            {
+                                uint32_t value;
+                                char const* name;
+                            }
+                            attribute_target_enums[] =
+                            {
+                                { 1, "Delegate" },
+                                { 2, "Enum" },
+                                { 4, "Event" },
+                                { 8, "Field" },
+                                { 16, "Interface" },
+                                { 64, "Method" },
+                                { 128, "Parameter" },
+                                { 256, "Property" },
+                                { 512, "Class" },   // "RuntimeClass"
+                                { 1024, "Struct" },
+                                { 2048, "All" },    // "InterfaceImpl"
```

Comment:
Are we saying the target is all for InterfaceImpl?

Reply:
this is existing code